library/alloc/src/collections/vec_deque/mod.rs

```diff
@@ -1336,7 +1336,8 @@ impl<T> VecDeque<T> {
     /// ```
     #[stable(feature = "rust1", since = "1.0.0")]
     pub fn back(&self) -> Option<&T> {
-        if !self.is_empty() { Some(&self[self.len() - 1]) } else { None }
+        let last = self.len() - 1;
```

Comment:
These `last` variables really don't add anything if you ask me. I'd just drop them.